src/generic/datavgen.cpp

```diff
@@ -3192,6 +3356,303 @@ int wxDataViewMainWindow::GetLineHeight( unsigned int row ) const
 }
 
 
+void RowRange::Add(const unsigned int idx)
+{
+    RangeEntity *rng = NULL;
+
+    size_t count = m_ranges.GetCount();
+    size_t rngIdx = 0;
+    while (rngIdx < count)
+    {
+        rng = &m_ranges[rngIdx];
+
+        if (idx >= rng->from && rng->to >= idx)
+        {
+            // index already in range
+            return;
+        }
+
+        if (rng->from == idx + 1)
+        {
+            rng->from = idx;
+            CleanUp(rngIdx);
+            return;
+        }
+        if (rng->to == idx - 1)
+        {
+            rng->to = idx;
+            CleanUp(rngIdx);
+            return;
+        }
+
+        if (rng->from > idx + 1)
+        {
+            break;
+        }
+
+        rngIdx += 1;
+    }
+//    wxLogMessage("New Range: %d" , count);
+
+    RangeEntity newRange;
+    newRange.from = idx;
+    newRange.to = idx;
+    m_ranges.Insert(newRange, rngIdx);
+}
+
+void RowRange::RemoveFrom(const unsigned int idx)
+{
+    RangeEntity *rng = NULL;
```

Comment:
Same remarks as above: please prefer references to pointers and `for` to `while` in this function too.